### DISCUSSION TITLE
Form options

### DISCUSSION THREAD

**Discussion by evamueller** [stephanrauh/ngx-extended-pdf-viewer#1262] [Q&A]
Hello together,

is there a way to check if the shown pdf has form elements?

And if it has form elements, is it possible to disable and enable them on the fly?

Thank you for your answers!

**Comment by stephanrauh** [comment]
If the PDF file is small enough, you can use a CSS selector like `document.querySelectorAll(".pdfViewer input")`. There's a event called `(textLayerRendered)`. You can use it to disable form field immediately after rendering them.

Larger PDF files aren't rendered completely at load-time. You can access the low-level API to check the annotation storage. In pure JavaScript, that's `PDFViewerApplication.pdfDocument.annotationStorage`. Also check out the type definitions (https://github.com/stephanrauh/ngx-extended-pdf-viewer/tree/main/projects/ngx-extended-pdf-viewer/types) to get an overview over the data structures.
